conans/client/conf/config_installer.py

```diff
@@ -0,0 +1,116 @@
+import os
+from conans.tools import unzip
+import shutil
+from conans.util.files import rmdir, load, mkdir
+from conans.client.remote_registry import RemoteRegistry, Remote
+from conans import tools
+from conans.errors import ConanException
+
+
+def _handle_remotes(registry_path, remote_file, output):
+    registry = RemoteRegistry(registry_path, output)
+
+    # Parse the remotes info
+    new_remotes = []
+    lines = [line.strip() for line in load(remote_file).splitlines() if line.strip()]
+    for line in lines:
+        name, url, ssl = line.split()
+        output.info("    Defining remote %s" % name)
+        new_remotes.append(Remote(name, url, ssl))
+
+    registry.define_remotes(new_remotes)
+
+
+def _handle_profiles(source_folder, target_folder, output):
+    mkdir(target_folder)
+    for root, _, files in os.walk(source_folder):
+        relative_path = os.path.relpath(root, source_folder)
+        if relative_path == ".":
+            relative_path = ""
+        for f in files:
+            profile = os.path.join(relative_path, f)
+            output.info("    Installing profile %s" % profile)
+            shutil.copy(os.path.join(root, f), os.path.join(target_folder, profile))
+
+
+def _process_git_repo(repo_url, client_cache, output, runner, tmp_folder):
+    output.info("Trying to clone repo  %s" % repo_url)
+
+    with tools.chdir(tmp_folder):
+        runner('git clone "%s" config' % repo_url, output=output)
+    tmp_folder = os.path.join(tmp_folder, "config")
+    _process_folder(tmp_folder, client_cache, output)
+
+
+def _process_zip_file(zippath, client_cache, output, tmp_folder):
+    unzip(zippath, tmp_folder)
+    os.unlink(zippath)
+    _process_folder(tmp_folder, client_cache, output)
+
+
+def _handle_conan_conf(current_conan_conf, new_conan_conf_path):
+    current_conan_conf.read(new_conan_conf_path)
+    with open(current_conan_conf.filename, "w") as f:
+        current_conan_conf.write(f)
+
+
+def _process_folder(folder, client_cache, output):
+    for root, dirs, files in os.walk(folder):
```

Comment:
a file "subdir1/subdir2/settings.yml" will be valid too. Why a walk and not just a listdir()?

Reply:
But ``listdir()`` only returns you the first level folder, so it will return ``subdir1`` only, and you will miss there is a settings.yml inside

Reply:
That is the problem. Do we want a `subdir/settings.yml` being valid? and a zip with two settings.yml?

Reply:
Yes, that is a good point, but I didn't want to restrict the layout, just in case someone wants to opt for a different one...

Regarding the os.walk() in any case the file copying will be done, so for example if they want to provide python files, the pylintrc, etc in other folder not the root one, they can do it.